### PR TITLE
Add toggle example showing how to combine with StartEndCommand

### LINKED ISSUE #969: Add toggle example showing how to combine with StartEndCommand
Many users want to use toggleWhenPressed to run two different commands.  It is not immediately obvious that StartEndCommand or ConditionalCommand may be combined with it to enable this functionality.

### PR DESCRIPTION
**Pull request by jasondaming**
Closes #969

### CODE REVIEW

File: source/docs/software/commandbased/binding-commands-to-triggers.rst [hunk 1]
```diff
@@ -37,7 +37,21 @@ This binding schedules a command when a trigger changes from active to inactive
...
+
+    myButton.toggleWhenPressed(new StartEndCommand(mySubsystem::onMethod,
+        mySubsystem::offMethod,
+        mySubsystem));
+
+  .. code-tab:: c++
+
+    myButton.ToggleWhenPressed(StartEndCommand([&] { mySubsystem.OnMethod(); },
+        [&] { mySubsystem.OffMethod(); },
+        mySubsystem));
```
Comment (prateekma) [inline source/docs/software/commandbased/binding-commands-to-triggers.rst:54]:
```suggestion
        {mySubsystem}));
```

**Review by prateekma — CHANGES_REQUESTED** [review]

File: source/docs/software/commandbased/binding-commands-to-triggers.rst (hunk 1 above)
Comment (jasondaming) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst:54]: once again never trust @calcmogul :)

File: source/docs/software/commandbased/binding-commands-to-triggers.rst [hunk 2]
```diff
@@ -37,7 +37,21 @@ This binding schedules a command when a trigger changes from active to inactive
 toggleWhenActive/toggleWhenPressed
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-This binding toggles a command, scheduling it when a trigger changes from inactive to active (or a button is initially pressed), and canceling it under the same condition if the command is currently running. Note that while this functionality is supported, toggles are *not* a highly-recommended option for user control, as they require the driver to mentally keep track of the robot state.
+This binding toggles a command, scheduling it when a trigger changes from inactive to active (or a button is initially pressed), and canceling it under the same condition if the command is currently running. Note that while this functionality is supported, toggles are not a highly-recommended option for user control, as they require the driver to keep track of the robot state.  If possible using two buttons one to turn on and another to turn off is preferred.  Using a StartEndCommand or a ConditionalCommand is a good way to specify the commands that you want to be want to be toggled between.
```
Comment (Daltz333) [inline source/docs/software/commandbased/binding-commands-to-triggers.rst:40]: ``If possible using two buttons one to turn on and another to turn off is preferred.``
Seems like a run-on sentence.

Comment (Daltz333) [inline source/docs/software/commandbased/binding-commands-to-triggers.rst:40]:
```suggestion
This binding toggles a command, scheduling it when a trigger changes from inactive to active (or a button is initially pressed), and canceling it under the same condition if the command is currently running. Note that while this functionality is supported, toggles are not a highly-recommended option for user control, as they require the driver to keep track of the robot state.  If possible using two buttons one to turn on and another to turn off is preferred.  Using a StartEndCommand or a ConditionalCommand is a good way to specify the commands that you want to be want to be toggled between.
```

Can we link to the API docs for StartEndCommand and ConditionalCommand

File: source/docs/software/commandbased/binding-commands-to-triggers.rst [hunk 3]
```diff
@@ -37,7 +37,21 @@ This binding schedules a command when a trigger changes from active to inactive
...
+
+  .. code-tab:: java
+
+    myButton.toggleWhenPressed(new StartEndCommand(mySubsystem::onMethod,
+        mySubsystem::offMethod,
+        mySubsystem));
+
+  .. code-tab:: c++
+
+    myButton.ToggleWhenPressed(StartEndCommand([&] { mySubsystem.OnMethod(); },
```
Comment (Daltz333) [inline source/docs/software/commandbased/binding-commands-to-triggers.rst:52]: @calcmogul Can you verify that this is accurate? I know you shared it in discord, but does this compile/run?

**Review by Daltz333 — CHANGES_REQUESTED** [review]

File: source/docs/software/commandbased/binding-commands-to-triggers.rst (hunk 2 above)
Comment (jasondaming) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst:40]: I am not the best at grammar but I think you are right and it should be: If possible using two buttons, one to turn on and another to turn off, is preferred.  Look better?

File: source/docs/software/commandbased/binding-commands-to-triggers.rst (hunk 3 above)
Comment (jasondaming) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst:52]: I am just gonna let him an @prateekma hash it out I don't know C++ well enough

File: source/docs/software/commandbased/binding-commands-to-triggers.rst (hunk 2 above)
Comment (Daltz333) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst:40]: What about "The preferred method is to use two buttons; one to turn on and another to turn off"

Comment (jasondaming) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst]: done

File: source/docs/software/commandbased/binding-commands-to-triggers.rst (hunk 1 above)
Comment (calcmogul) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst]: Bruh. lol

File: source/docs/software/commandbased/binding-commands-to-triggers.rst (hunk 2 above)
Comment (jasondaming) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst]: Sold!

File: source/docs/software/commandbased/binding-commands-to-triggers.rst (hunk 3 above)
Comment (calcmogul) [reply source/docs/software/commandbased/binding-commands-to-triggers.rst:52]: Seems correct based on the function signature at least.

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**